rllib/policy/eager_tf_policy_v2.py

```diff
@@ -0,0 +1,880 @@
+    def optimizer(self) -> Union["tf.keras.optimizers.Optimizer",
+                                 List["tf.keras.optimizers.Optimizer"]]:
+        """TF optimizer to use for policy optimization.
+
+        Returns:
+            A local optimizer or a list of local optimizers to use for this
+                Policy's Model.
+        """
+        return tf.keras.optimizers.Adam(self.config["lr"])
+
+    def maybe_initialize_optimizer_and_loss(self):
+        optimizers = force_list(self.optimizer())
+        if getattr(self, "exploration", None):
+            optimizers = self.exploration.get_exploration_optimizer(optimizers)
+
+        # The list of local (tf) optimizers (one per loss term).
+        self._optimizers: List[LocalOptimizer] = optimizers
+        # Backward compatibility: A user's policy may only support a single
+        # loss term and optimizer (no lists).
+        self._optimizer: LocalOptimizer = optimizers[0] if optimizers else None
+
+        self._initialize_loss_from_dummy_batch(auto_remove_unneeded_view_reqs=True,)
+        self._loss_initialized = True
+
+    @override(Policy)
+    def compute_actions_from_input_dict(
+        self,
+        input_dict: Dict[str, TensorType],
+        explore: bool = None,
+        timestep: Optional[int] = None,
+        episodes: Optional[List[Episode]] = None,
+        **kwargs,
+    ) -> Tuple[TensorType, List[TensorType], Dict[str, TensorType]]:
+
+        if not self.config.get("eager_tracing") and not tf1.executing_eagerly():
+            tf1.enable_eager_execution()
+
+        self._is_training = False
+
+        explore = explore if explore is not None else self.explore
+        timestep = timestep if timestep is not None else self.global_timestep
+        if isinstance(timestep, tf.Tensor):
+            timestep = int(timestep.numpy())
+
+        # Pass lazy (eager) tensor dict to Model as `input_dict`.
+        input_dict = self._lazy_tensor_dict(input_dict)
+        input_dict.set_training(False)
+
+        # Pack internal state inputs into (separate) list.
+        state_batches = [
+            input_dict[k] for k in input_dict.keys() if "state_in" in k[:8]
+        ]
+        self._state_in = state_batches
+        self._is_recurrent = state_batches != []
+
+        # Call the exploration before_compute_actions hook.
+        self.exploration.before_compute_actions(
+            timestep=timestep, explore=explore, tf_sess=self.get_session()
+        )
+
+        ret = self._compute_actions_helper(
+            input_dict,
+            state_batches,
+            # TODO: Passing episodes into a traced method does not work.
+            None if self.config["eager_tracing"] else episodes,
+            explore,
+            timestep,
+        )
+        # Update our global timestep by the batch size.
+        self.global_timestep.assign_add(tree.flatten(ret[0])[0].shape.as_list()[0])
+        return convert_to_numpy(ret)
+
+    @override(Policy)
+    def compute_actions(
```

Comment:
Can we deprecate this entirely as the class is new anyways?
RLlib should only use `compute_actions_from_input_dict`.

Also, this is probably a very good time to cleanup the Policy API in general. Let's discuss this internally. I guess this is the same as the already ongoing Ray AIR discussion :)

Reply:
yeah, sounds like a good idea.
we need to do that after all the agents are migrated to the new policy though.
I am sure we have a lot of code/examples that are still using this API call.